fastfuels_sdk/exports.py

```diff
@@ -313,6 +326,336 @@ def export_zarr_to_fds(zroot: zarr.hierarchy.Group,
+        if fuel_types == "total":
+            raise Warning("All fuel types will be calibrated when keyword argument fuel type is 'total'. Consider removing keep_duet argument. Proceeding with calibration.")
+        
+    # Print function specifications
+
+    if fuel_types == ["none"]:
+        grass_print = "Landfire SB40"
+        litter_print = "Landfire SB40"
+    elif fuel_types in [["total"],["grass","litter"],["litter","grass"]]:
+        grass_print = "User inputs"
+        litter_print = "User inputs"
+    elif fuel_types == ["grass"]:
+        grass_print = "User inputs"
+        litter_print = "Landfire SB40"
+    elif fuel_types == ["litter"]:
+        grass_print = "Landfire SB40"
+        litter_print = "User inputs"
+    
+    if keep_duet == "grass":
+        grass_print = "Keep DUET values"
+    elif keep_duet == "litter":
+        litter_print = "Keep DUET values"
+    
+    if grass_print == "Landfire SB40":
+        grass_method = "max/min"
+    elif ([key for key in kwargs.keys() if key.endswith("mean") or key.endswith("sd")]):
+        grass_method = "mean/sd"
+    elif ([key for key in kwargs.keys() if key.endswith("max") or key.endswith("min")]):
+        grass_method = "max/min"
+    if litter_print == "Landfire SB40":
+        litter_method = "max/min"
+    elif ([key for key in kwargs.keys() if key.endswith("mean") or key.endswith("sd")]):
+        litter_method = "mean/sd"
+    elif ([key for key in kwargs.keys() if key.endswith("max") or key.endswith("min")]):
+        litter_method = "max/min"
+    
+    
+    print("\nCalibration Specifications:")
+    print("GRASS: {} ({})".format(grass_print,grass_method))
+    print("LITTER: {} ({})\n".format(litter_print,litter_method))
+
+
+    # If user inputs are not present for all fuel types, use values from Landfire:
+    if fuel_types == ["litter"] and keep_duet=="grass":
+        pass
+    elif fuel_types == ["grass"] and keep_duet=="litter":
+        pass
+    elif fuel_types == ["litter"] or fuel_types == ["grass"] or fuel_types == ["none"]:
+        print("Querying LandFire...\n")
+        # Query Landfire and return array of SB40 keys
+        sb40_arr = _query_landfire(zroot, output_dir)
+        # Import SB40 FBFM parameters table
+        if param_dir == None:
+            param_dir = output_dir
+        sb40_params = pd.read_csv(Path(param_dir,"sb40_parameters.csv"))
+
+        # Generate dict of fastfuels bulk density values and apply to Landfire query
+        sb40_dict = _get_sb40_fuel_params(sb40_params)
+        ftype_arr, rhof_arr = _get_sb40_arrays(sb40_arr, sb40_dict)
+    
+    # Read in bulk density output from DUET
+    nx = zroot.attrs["nx"]
+    ny = zroot.attrs["ny"]
+    nz = 2 # number of duet layers, right now grass and litter. Will be number of tree species + 1
+    dim = (nz,ny,nx)
+    duet_rhof = _read_dat_file(output_dir, "surface_rhof.dat", dim, order="F")
+    # Calibrate based on fuel_types and kwargs
```

Comment:
What would you think of replacing some of the redunacny in the logic with helper functions like so:

```python
def get_stat_values(fuel_type, kwargs):
    mean_key = f"{fuel_type}_mean"
    sd_key = f"{fuel_type}_sd"
    max_key = f"{fuel_type}_max"
    min_key = f"{fuel_type}_min"
    
    return kwargs.get(mean_key), kwargs.get(sd_key), kwargs.get(max_key), kwargs.get(min_key)

def calibrate_by_fuel_type(fuel_type, data, kwargs):
    mean, sd, max_val, min_val = get_stat_values(fuel_type, kwargs)

    if mean and sd:
        return _calibrate_meansd(data, mean, sd)
    elif max_val and min_val:
        return _calibrate_maxmin(data, max_val, min_val)
    else:
        return data

# Main Calibration Logic
if fuel_types == ["total"]:
    duet_rhof = np.add(duet_rhof[0,:,:], duet_rhof[1,:,:])
    duet_calibrated = calibrate_by_fuel_type('total', duet_rhof, kwargs)
else:
    duet_grass = duet_rhof[0,:,:]
    duet_litter = duet_rhof[1,:,:]
    
    grass_calibrated = calibrate_by_fuel_type('grass', duet_grass, kwargs)
    litter_calibrated = calibrate_by_fuel_type('litter', duet_litter, kwargs)

    # Additional logic for handling `keep_duet` and combining results can be added here
    duet_calibrated = np.add(grass_calibrated, litter_calibrated) 
```